src/main/java/net/minecraftforge/network/PlayMessages.java

```diff
@@ -122,42 +132,67 @@ public static SpawnEntity decode(FriendlyByteBuf buf)
                     buf.readDouble(), buf.readDouble(), buf.readDouble(),
                     buf.readByte(), buf.readByte(), buf.readByte(),
                     buf.readShort(), buf.readShort(), buf.readShort(),
-                    buf
-                    );
+                    readSpawnDataPacket(buf)
+            );
         }
 
-        public static void handle(SpawnEntity msg, Supplier<NetworkEvent.Context> ctx)
+        private static FriendlyByteBuf readSpawnDataPacket(FriendlyByteBuf buf)
         {
-            ctx.get().enqueueWork(() -> {
-                EntityType<?> type = Registry.ENTITY_TYPE.byId(msg.typeId);
-                if (type == null)
-                {
-                    throw new RuntimeException(String.format(Locale.ENGLISH, "Could not spawn entity (id %d) with unknown type at (%f, %f, %f)", msg.entityId, msg.posX, msg.posY, msg.posZ));
-                }
+            final int count = buf.readVarInt();
+            if (count > 0)
+            {
+                final FriendlyByteBuf spawnDataBuffer = new FriendlyByteBuf(Unpooled.buffer());
+                spawnDataBuffer.writeBytes(buf, count);
+                return spawnDataBuffer;
+            }
 
-                Optional<Level> world = LogicalSidedProvider.CLIENTWORLD.get(ctx.get().getDirection().getReceptionSide());
-                Entity e = world.map(w->type.customClientSpawn(msg, w)).orElse(null);
-                if (e == null)
-                {
-                    return;
-                }
+            return new FriendlyByteBuf(Unpooled.buffer());
+        }
 
-                /*
-                 * Sets the postiion on the client, Mirrors what
-                 * Entity#recreateFromPacket and LivingEntity#recreateFromPacket does.
-                 */
-                e.syncPacketPositionCodec(msg.posX, msg.posY, msg.posZ);
-                e.absMoveTo(msg.posX, msg.posY, msg.posZ, (msg.yaw * 360) / 256.0F, (msg.pitch * 360) / 256.0F);
-                e.setYHeadRot((msg.headYaw * 360) / 256.0F);
-                e.setYBodyRot((msg.headYaw * 360) / 256.0F);
-
-                e.setId(msg.entityId);
-                e.setUUID(msg.uuid);
-                world.filter(ClientLevel.class::isInstance).ifPresent(w->((ClientLevel)w).putNonPlayerEntity(msg.entityId, e));
-                e.lerpMotion(msg.velX / 8000.0, msg.velY / 8000.0, msg.velZ / 8000.0);
-                if (e instanceof IEntityAdditionalSpawnData)
+        public static void handle(SpawnEntity msg, Supplier<NetworkEvent.Context> ctx)
+        {
+            ctx.get().enqueueWork(() ->
```

Comment:
I beg you to make this function easier to read.

I legitimately cannot parse the control flow or order of parameters being used here.

Please make things one giant line if it benefits, this ridiculous line splitting is horrendous for the rest of us.